Fix Fleet-27 test case failure behavior

sbulage opened the pull request:
Fixes: #215

sbulage commented:
:gear:  2.8 CI RUN: https://github.com/rancher/fleet-e2e/actions/runs/11250486524

mmartin24 commented on tests/cypress/support/commands.ts:
```diff
@@ -557,6 +557,9 @@ Cypress.Commands.add('removeClusterLabels', (clusterName, key, value) => {
   cy.wait(500);
   cy.clickButton('Save');
   cy.contains('Save').should('not.exist');
+  // After label removal from cluster, it says 409 (conflict error) while saving.
```
Has this been reported as a bug?

sbulage replied:
I would say, there is no as such bug for this behavior.

mmartin24 replied:
As you found out it seems there was and I commented there: `https://github.com/rancher/dashboard/issues/9563#issuecomment-2404603657`

If this workaround helps I am ok, but pls add a comment to remove it once the issue is fixed.

mmartin24 replied:
Also, tests failing non-related to this, I guess is a hiccup? Could you pls ensure other tests are not affected (77,20,30,31 etc?). Thanks.

sbulage replied:
The tests which are failing 30,31 are related to this test failure. But the 77,20 are not related could be a hiccup.

mmartin24 commented on tests/cypress/support/commands.ts:
```diff
@@ -551,10 +551,32 @@ Cypress.Commands.add('removeClusterLabels', (clusterName, key, value) => {
+      let disabled_button = cy.get('div[class="row"] div[class="key-value"] button.role-link')
+
+      if (disabled_button < 3) {
+        cy.log("working");
```
Maybe leftover?
Probably better something like "Only 2 default label. Nothing to do"

sbulage replied:
Oops, I just saw the comment but I updated with different log message. I will update it with your suggested changes in next patch set.

sbulage commented:
CI 2.10.0 Run: https://github.com/rancher/fleet-e2e/actions/runs/11725723468

mmartin24 commented:
There are failures both in 2.10 and 2.9. 
I am set as reviewer. Is this ready for review?

sbulage commented:
> There are failures both in 2.10 and 2.9. I am set as reviewer. Is this ready for review?

You're set as a reviewer, but it is not yet ready for review. It is _In Progress_ on board now.
I will ping you once it is ready for review.

mmartin24 commented:
> > There are failures both in 2.10 and 2.9. I am set as reviewer. Is this ready for review?
> 
> You're set as a reviewer, but it is not yet ready for review. It is _In Progress_ on board now. I will ping you once it is ready for review.

ok, it was not clear for me since I was pinged in Slack with updates on it. Normally we do so / set for review once pr is ready, but sure, now I know no action is needed on my side until I am pinged directly with call to action.

sbulage commented:
> > > There are failures both in 2.10 and 2.9. I am set as reviewer. Is this ready for review?
> > 
> > 
> > You're set as a reviewer, but it is not yet ready for review. It is _In Progress_ on board now. I will ping you once it is ready for review.
> 
> ok, it was not clear for me since I was pinged in Slack with updates on it. Normally we do so / set for review once pr is ready, but sure, now I know no action is needed on my side until I am pinged directly with call to action.

Yes, I can understand it. But I have added you as a reviewer at that time we saw that it was fixed but later onwards, I was moved to release testing. So this PR was hanged for that moment. Later, we saw few test cases were failing, so I started working on it again.

I changed the status of this PR from In Review to In Progress. On Slack, I gave you update that what is not working if I add the workaround and currently I was working on fix.

I apologize if you get confused due to this PR. For the time being, please don't review this PR until I change the status from In Progress to In Review.

mmartin24 commented:
> > > > There are failures both in 2.10 and 2.9. I am set as reviewer. Is this ready for review?
> > > 
> > > 
> > > You're set as a reviewer, but it is not yet ready for review. It is _In Progress_ on board now. I will ping you once it is ready for review.
> > 
> > 
> > ok, it was not clear for me since I was pinged in Slack with updates on it. Normally we do so / set for review once pr is ready, but sure, now I know no action is needed on my side until I am pinged directly with call to action.
> 
> Yes, I can understand it. But I have added you as a reviewer at that time we saw that it was fixed but later onwards, I was moved to release testing. So this PR was hanged for that moment. Later, we saw few test cases were failing, so I started working on it again.
> 
> I changed the status of this PR from In Review to In Progress. On Slack, I gave you update that what is not working if I add the workaround and currently I was working on fix.
> 
> I apologize if you get confused due to this PR. For the time being, please don't review this PR until I change the status from In Progress to In Review.

Just to clarify, I never saw this fixed. I said this was a workaround and commented [here](https://github.com/rancher/fleet-e2e/pull/216#discussion_r1796525426) if you think it works go try, but that's all.

sbulage commented:
> > > > > There are failures both in 2.10 and 2.9. I am set as reviewer. Is this ready for review?
> > > > 
> > > > 
> > > > You're set as a reviewer, but it is not yet ready for review. It is _In Progress_ on board now. I will ping you once it is ready for review.
> > > 
> > > 
> > > ok, it was not clear for me since I was pinged in Slack with updates on it. Normally we do so / set for review once pr is ready, but sure, now I know no action is needed on my side until I am pinged directly with call to action.
> > 
> > 
> > Yes, I can understand it. But I have added you as a reviewer at that time we saw that it was fixed but later onwards, I was moved to release testing. So this PR was hanged for that moment. Later, we saw few test cases were failing, so I started working on it again.
> > I changed the status of this PR from In Review to In Progress. On Slack, I gave you update that what is not working if I add the workaround and currently I was working on fix.
> > I apologize if you get confused due to this PR. For the time being, please don't review this PR until I change the status from In Progress to In Review.
> 
> Just to clarify, I never saw this fixed. I said this was a workaround and commented [here](https://github.com/rancher/fleet-e2e/pull/216#discussion_r1796525426) if you think it works go try, but that's all.

Yes, you're correct, it's a workaround until UI issue gets fixed.

sbulage commented:
CI 2.10.0 Run: https://github.com/rancher/fleet-e2e/actions/runs/11739338191

sbulage commented:
2.9 all test passed which is fixed by this PR:

![image](https://github.com/user-attachments/assets/647b735a-4577-4c83-b4e8-44ca18c23492)

sbulage commented:
:green_circle: All 2.10.0-->  P1 tests passed: https://github.com/rancher/fleet-e2e/actions/runs/11743323387

sbulage commented:
Another 2.10.0 CI Run: https://github.com/rancher/fleet-e2e/actions/runs/11774095706 <-- :green_circle:

mmartin24 commented:
> 2.9 all test passed which is fixed by this PR:
> 
> ![image](https://private-user-images.githubusercontent.com/12134754/384447270-647b735a-4577-4c83-b4e8-44ca18c23492.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzOTk5MjUsIm5iZiI6MTczMTM5OTYyNSwicGF0aCI6Ii8xMjEzNDc1NC8zODQ0NDcyNzAtNjQ3YjczNWEtNDU3Ny00YzgzLWI0ZTgtNDRjYTE4YzIzNDkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEyVDA4MjAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzYzFhNzlhYWM4NGQyMWU3OGExYWRhNTM2OGI4Y2Y1MzYyMjk2YjlkZWNiOTUxYmZhNDA3ZmVmOTEzNTQyYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.s93Y0g25HxcUXj9OsbHvBrWY_UV7muvE9XUJ8CZs5kE)

FYI: for the past 4 days this issue is not occurring in 2.9 in main, so the actual fix may have occurred there: [695](https://github.com/rancher/fleet-e2e/actions/runs/11753188174/job/32745765497#step:9:424),[696](https://github.com/rancher/fleet-e2e/actions/runs/11762062267/job/32764556223#step:9:406),[697](https://github.com/rancher/fleet-e2e/actions/runs/11771954543/job/32788795685#step:9:421),[698](https://github.com/rancher/fleet-e2e/actions/runs/11790557261/job/32844622239#step:9:418). The implementation of the cleanup at first it is though an important addition.

In 2.10, still fails so it is nice if the workaround on this pr may have an impact there. Happy to review it when ready.

sbulage commented:
Yes, it is Ready for Review :white_check_mark:

mmartin24 commented:
ok. looks good. Can you pls take a look at this run on 2.8. It fails there: https://github.com/rancher/fleet-e2e/actions/runs/11795263567/job/32854656095
maybe some locator adjustments?

sbulage commented:
CI 2.8 Run: https://github.com/rancher/fleet-e2e/actions/runs/11797775737
CI 2.10.0 Run: https://github.com/rancher/fleet-e2e/actions/runs/11797784237
